data/crac/crac-loopflow-extension/src/main/java/com/farao_community/farao/data/crac_loopflow_extension/CnecLoopFlowExtension.java

```diff
@@ -77,4 +91,76 @@ public boolean hasLoopflowShift() {
     public void setHasLoopflowShift(boolean hasLoopflowShift) {
         this.hasLoopflowShift = hasLoopflowShift;
     }
+
+    public double getInputThreshold(Unit requestedUnit, Network network) {
+
+        if (requestedUnit.getPhysicalParameter() != PhysicalParameter.FLOW) {
+            throw new FaraoException("Loop thresholds can only be returned in AMPERE, MEGAWATT or PERCENT_IMAX");
+        }
+
+        if (requestedUnit == inputThresholdUnit) {
+            return inputThreshold;
+        }
+
+        if (inputThresholdUnit == Unit.PERCENT_IMAX && requestedUnit == Unit.AMPERE) {
+            return convertPercentImaxToA(network, inputThreshold);
+        }
+
+        if (inputThresholdUnit == Unit.PERCENT_IMAX && requestedUnit == Unit.MEGAWATT) {
+            return convertAToMW(network, convertPercentImaxToA(network, inputThreshold));
+        }
+
+        if (inputThresholdUnit == Unit.AMPERE && requestedUnit == Unit.PERCENT_IMAX) {
+            return convertAToPercentImax(network, inputThreshold);
+        }
+
+        if (inputThresholdUnit == Unit.AMPERE && requestedUnit == Unit.MEGAWATT) {
+            return convertAToMW(network, inputThreshold);
+        }
+
+        if (inputThresholdUnit == Unit.MEGAWATT && requestedUnit == Unit.AMPERE) {
+            return convertMWToA(network, inputThreshold);
+        }
+
+        if (inputThresholdUnit == Unit.MEGAWATT && requestedUnit == Unit.PERCENT_IMAX) {
+            return convertAToPercentImax(network, convertMWToA(network, inputThreshold));
+        }
+
+        throw new FaraoException(String.format("Cannot convert %s into %s", inputThresholdUnit, requestedUnit));
+
+    }
+
+    @Override
+    public String getName() {
+        return "CnecLoopFlowExtension";
+    }
+
+    private double convertMWToA(Network network, double valueInMW) {
+        return valueInMW * 1000 / (getNominalVoltage(network) * Math.sqrt(3));
+    }
+
+    private double convertAToMW(Network network, double valueInA) {
+        return valueInA * getNominalVoltage(network) * Math.sqrt(3) / 1000;
+    }
+
+    private double convertAToPercentImax(Network network, double valueInA) {
+        return valueInA * 100 / getCnecThresholdInA(network);
+    }
+
+    private double convertPercentImaxToA(Network network, double valueInPercent) {
+        return valueInPercent * getCnecThresholdInA(network) / 100;
+    }
+
+    private double getNominalVoltage(Network network) {
+        Branch branch = network.getBranch(this.getExtendable().getNetworkElement().getId());
+        if (branch == null) {
+            throw new FaraoException(String.format("Cnec with id %s was not found in the network", this.getExtendable().getId()));
+        }
+        return branch.getTerminal1().getVoltageLevel().getNominalV();
+    }
+
+    private double getCnecThresholdInA(Network network) {
+        return Math.min(getExtendable().getMaxThreshold(Unit.AMPERE).orElse(Double.POSITIVE_INFINITY),
```

Comment:
- Isn't it -FRM and not +FRM ? (actually I'm not even sure you should consider FRM here as long as you want to compute Imax)

- What if we have infinite values here ? Isn't going to fail ? Shouldn't we ensure that the threshold final value is not infinite ?

Reply:
I am adding a check in the maxLoopFlowFiller to ensure that we do not build a constraint if the threshold is inifinite

The threshold would only be infinite if the Cnec has no threshold.